src/Adapter/MSTest.CoreAdapter/Execution/UnitTestRunner.cs

```diff
@@ -80,17 +87,19 @@ internal UnitTestResult[] RunSingleTest(TestMethod testMethod, IDictionary<strin
                     testContext.SetOutcome(TestTools.UnitTesting.UnitTestOutcome.InProgress);
 
                     // Get the testMethod
-                    var testMethodInfo = this.typeCache.GetTestMethodInfo(testMethod, testContext, MSTestSettings.CurrentSettings.CaptureDebugTraces);
+                    var testMethodInfo = this.typeCache.GetTestMethodInfo(testMethod, testContext, testExecutionRecorder, MSTestSettings.CurrentSettings.CaptureDebugTraces);
 
                     // If the specified TestMethod could not be found, return a NotFound result.
                     if (testMethodInfo == null)
                     {
+                        testExecutionRecorder.RecordStartAndEnd(new UnitTestElement(testMethod), UnitTestOutcome.NotFound);
```

Comment:
I am not able to understand the use case for this.

Reply:
Same for this class. Not required to record here.